src/resolvers/workspace.js

```diff
@@ -319,6 +320,61 @@ module.exports = {
       return true;
     },
 
+    /**
+     * Mutation in order to leave workspace
+     * @param {ResolverObj} _obj - object that contains the result returned from the resolver on the parent field
+     * @param {string} workspaceId - id of the workspace where the user should be removed
+     * @param {UserInContext} user - current authorized user {@see ../index.js}
+     * @param {ContextFactories} factories - factories for working with models
+     * @return {Promise<boolean>} - true if operation is successful
+     */
+    async deleteWorkspace(_obj, { workspaceId }, { user, factories }) {
+      const workspaceModel = await factories.workspacesFactory.findById(workspaceId);
+
+      if (!workspaceModel) {
+        throw new UserInputError('There is no workspace with provided id');
+      }
+
+      const memberInfo = await workspaceModel.getMemberInfo(user.id);
+
+      if (memberInfo.isAdmin) {
+        const membersInfo = (await workspaceModel.getMembers());
+
+        for (const member of membersInfo) {
+          const userModel = await factories.usersFactory.findById(member.userId.toString());
+
+          await workspaceModel.removeMember(userModel);
```

Comment:
why do we need to remove members one-by-one, if you are dropping the whole team collection in `workspaceModel.deleteWorkspace();`?

Reply:
we need to remove the workspace from the user also that why which is handle by removeMember